File: conference/payments-bc/web/src/main/java/com/microsoft/conference/payments/messagepublishers/PaymentMessagePublisher.cs
using System.Threading.Tasks;
using ECommon.Components;
using ECommon.IO;
using ENode.Infrastructure;
using Payments.Messages;

namespace Payments.MessagePublishers
{
    [Component]
    public class PaymentMessagePublisher :
        IMessageHandler<PaymentCompleted>,
        IMessageHandler<PaymentRejected>
    {
        private readonly IMessagePublisher<IApplicationMessage> _messagePublisher;

        public PaymentMessagePublisher(IMessagePublisher<IApplicationMessage> messagePublisher)
        {
            _messagePublisher = messagePublisher;
        }

        public Task<AsyncTaskResult> HandleAsync(PaymentCompleted evnt)
        {
            return _messagePublisher.PublishAsync(new PaymentCompletedMessage
            {
                PaymentId = evnt.AggregateRootId,
                ConferenceId = evnt.ConferenceId,
                OrderId = evnt.OrderId
            });
        }
        public Task<AsyncTaskResult> HandleAsync(PaymentRejected evnt)
        {
            return _messagePublisher.PublishAsync(new PaymentRejectedMessage
            {
                PaymentId = evnt.AggregateRootId,
                ConferenceId = evnt.ConferenceId,
                OrderId = evnt.OrderId
            });
        }
    }
}

File: samples/conference/payments-bc/src/main/java/com/microsoft/conference/payments/messagepublishers/PaymentMessagePublisher.java
package com.microsoft.conference.payments.messagepublishers;

import com.microsoft.conference.common.payment.message.PaymentCompletedMessage;
import com.microsoft.conference.common.payment.message.PaymentRejectedMessage;
import com.microsoft.conference.payments.domain.events.PaymentCompleted;
import com.microsoft.conference.payments.domain.events.PaymentRejected;
import org.enodeframework.annotation.Event;
import org.enodeframework.annotation.Subscribe;
import org.enodeframework.messaging.IApplicationMessage;
import org.enodeframework.messaging.IMessagePublisher;
import org.springframework.beans.factory.annotation.Autowired;

import static org.enodeframework.common.io.Task.await;

@Event
public class PaymentMessagePublisher {

    @Autowired
    private IMessagePublisher<IApplicationMessage> messagePublisher;

    public PaymentMessagePublisher(IMessagePublisher<IApplicationMessage> messagePublisher) {
        this.messagePublisher = messagePublisher;
    }

    @Subscribe
    public void handleAsync(PaymentCompleted evnt) {
        PaymentCompletedMessage message = new PaymentCompletedMessage();
        message.paymentId = evnt.getAggregateRootId();
        message.conferenceId = evnt.getConferenceId();
        message.orderId = evnt.getOrderId();
        await(messagePublisher.publishAsync(message));
    }

    @Subscribe
    public void handleAsync(PaymentRejected evnt) {
        PaymentRejectedMessage message = new PaymentRejectedMessage();
        message.paymentId = evnt.getAggregateRootId();
        message.conferenceId = evnt.getConferenceId();
        message.orderId = evnt.getOrderId();
        await(messagePublisher.publishAsync(message));
    }
}
